transform/snowflake-dbt/models/events/nightly/user_events_by_date.sql

```diff
@@ -0,0 +1,346 @@
+           , e.category
+           , {{ dbt_utils.surrogate_key(['e.timestamp::date', 'e.user_actual_id', 'e.user_id', 'e.context_user_agent', 'lower(e.type)', 'e.category']) }}                       AS id
+         FROM {{ source('mattermost2', 'event') }} e
+              LEFT JOIN (
+                          SELECT user_id, MIN(TIMESTAMP::DATE) AS MIN_DATE
+                          FROM {{ source('mm_telemetry_prod', 'event') }}
+                          GROUP BY 1
+                        ) rudder
+                        ON e.timestamp::date >= rudder.MIN_DATE
+                        AND e.user_id = rudder.user_id
+         WHERE rudder.user_id IS NULL
+         AND e.timestamp::DATE < CURRENT_DATE
+         {% if is_incremental() %}
+
+          AND e.timestamp > (SELECT MAX(max_timestamp) from {{this}})
+
+         {% endif %}
+         GROUP BY 1, 2, 3, 5, 6, 7, 8, 9, 10, 12, 16, 17
+     ), 
+     events2 AS (
+         SELECT
+             e.timestamp::DATE                                                                         AS date
+           , TRIM(e.user_id)                                                                           AS server_id
+           , COALESCE(TRIM(e.user_actual_id), NULL)                                                    AS user_id
+           , min(e.user_actual_role)                                                                   AS user_role
+           , CASE
+              WHEN context_useragent LIKE '%iPhone%'    THEN 'iPhone'
+              WHEN context_useragent LIKE '%iPad%'      THEN 'iPad'
+              WHEN context_useragent LIKE '%Android%'   THEN 'Android'
+              WHEN context_useragent LIKE '%Electron/%' THEN 'Electron'
+              WHEN context_useragent LIKE '%Edge/%'     THEN 'Edge'
+              WHEN context_useragent LIKE '%Edg/%'      THEN 'Edge'
+              WHEN context_useragent LIKE '%MSIE%'      THEN 'IE'
+              WHEN context_useragent LIKE '%Trident/%'  THEN 'IE'
+              WHEN context_useragent LIKE '%Firefox/%'  THEN 'Firefox'
+              WHEN context_useragent LIKE '%Chrome/%'   THEN 'Chrome'
+              WHEN context_useragent LIKE '%Safari/%'   THEN 'Safari'
+              ELSE 'Other'
+              END                                                                                        AS browser
+           , CASE
+              WHEN context_useragent LIKE '%iPhone%'    THEN 'iPhone'
+              WHEN context_useragent LIKE '%iPad%'      THEN 'iPad'
+              WHEN context_useragent LIKE '%CrOS%'      THEN 'Chrome OS'
+              WHEN context_useragent LIKE '%Android%'   THEN 'Android'
+              WHEN context_useragent LIKE '%Macintosh%' THEN 'Mac'
+              WHEN context_useragent LIKE '%Windows%'   THEN 'Windows'
+              WHEN context_useragent LIKE '%Linux%'     THEN 'Linux'
```

Comment:
Is there really no better way to do this?

Reply:
cc @sri-byte a quick reminder on this comment, otherwise looks like this PR is ready to be merged